src/AlgebraicGeometry/Surfaces/EnriquesAut.jl

```diff
@@ -0,0 +1,944 @@
+  #upper_bound = mass(ECtx)*length(initial_automorphisms(ECtx))
+  return ECtx
+end
+  
+@doc raw"""
+    EnriquesBorcherdsCtx(SY2::ZZLat, SX::ZZLat) -> EnriquesBorcherdsCtx 
+  
+Return a context object for Borcherds' method for Enriques surfaces. 
+  
+Let ``\pi: X \to Y`` be the universal cover of an Enriques surfaces and ``\epsilon`` the Enriques involution. 
+# Input: 
+- `SY2` -- the invariant lattice of the Enriques involution in the numerical lattice `SX` of ``X``. 
+"""
+function EnriquesBorcherdsCtx(SY2::ZZLat, SX::ZZLat)
+  @req issubset(SY2, SX) "SY2 must be a sublattice of SX"
+  a = rescale(SY2,1//2)
+  @req is_unimodular(a) && is_even(a) && signature(a) == (1, 0, 9) "SY2 must be isomorphic to E10(2)"
+  B = solve(basis_matrix(SX), basis_matrix(SY2); side=:left)
+  L26, SX1, weyl1 = borcherds_method_preprocessing(S, 26)
+  SY1 = lattice(ambient_space(SX1), SY1)
+  return EnriquesBorcherdsCtx(SY1,SX1, L26, weyl1)
+end
+
+@doc raw"""
+    Enriques_surface_automorphism_group(SY2::ZZLat, SX::ZZLat)
+    
+Return generators for the automorphism group of an Enriques surface.
+  
+Let ``\pi: X \to Y`` be the universal cover of an Enriques surfaces and ``\epsilon`` the Enriques involution. Let ``S_Y`` be the numerical lattice of ``Y``. 
+This function computes the image of the natural map 
+```math
+\varphi_Y \colon \mathrm{Aut}_{s}(Y) \to O(S_Y \otimes \mathbb{F}_2)
+```
+where ``\mathrm{Aut}_{s}(Y) \leq \mathrm{Aut}(Y)`` denotes the subgroup of semi-symplectic automorphisms.
+Note that the kernel of ``\varphi_Y`` is finite.
+
+See [BS22](@cite) [BS22*1](@cite) [BRS23](@cite) for background and algorithms. 
+
+# Input: 
+- `SY2` -- the invariant lattice of the Enriques involution in the numerical lattice `SX` of ``X``. 
+"""
+function Enriques_surface_automorphism_group(SY2::ZZLat, SX::ZZLat)
+  return borcherds_method(EnriquesBorcherdsCtx(SY2::ZZLat, SX::ZZLat))
+end 
+  
+@doc raw"""
+    splitting_roots_mod2(Y::EnriquesBorcherdsCtx)
+    
+Return the image of the splitting roots of ``Y`` in ``S_Y \otimes \mathbb{F}_2``.
+"""
+function splitting_roots_mod2(Y::EnriquesBorcherdsCtx)
+  SX = Y.SX
+  SY2 = Y.SY
+  Sm = orthogonal_submodule(SX, SY2)
+  phi, inc_Dminus, inc_Dplus = glue_map(SX, Sm, SY2)
+  # H_Sm = pi_Sm(SX) note that H_Sm/Sm is
+  H_Sm = cover(domain(phi))
+  sv2 = [1//2*(i[1]*basis_matrix(Sm)) for i in short_vectors(rescale(Sm,-1),4) if i[2]==4]
+  sv2 = [domain(phi)(i) for i in sv2 if i in H_Sm]
+  return Set([change_base_ring(GF(2),solve(basis_matrix(SY2),matrix(QQ,1,26,2*lift(phi(i)));side=:left)) for i in sv2])
+end
+
+@doc raw"""
+    root_invariant(Y::EnriquesBorcherdsCtx)
+    
+Return the root invariant of ``Y``.
+"""
+function root_invariant(Y::EnriquesBorcherdsCtx)
+  Sm = orthogonal_submodule(Y.SX, Y.SY)
+  phi, inc_Dminus, inc_Dplus = glue_map(Y.SX, Sm, Y.SY)
+  # H_Sm = pi_Sm(SX)
+  H_Sm = cover(domain(phi))
+  sv2 = [1//2*(i[1]*basis_matrix(Sm)) for i in short_vectors(Sm, 4) if i[2]==4]
```

Comment:
```suggestion
  sv2 = [1//2*(i[1]*basis_matrix(Sm)) for i in short_vectors(Sm, 4) if i[2] == 4]
```
This set if computed in a lot of different places, in your preprocessing or in additional functions. Would not it make sense to store it ?